src/client/producer.rs

```diff
@@ -210,10 +210,10 @@ use crate::record::Record;
 pub mod aggregator;
 mod broadcast;
 
-#[derive(Debug, Error)]
+#[derive(Debug, Error, Clone)]
```

Comment:
Comment got orphaned by another change, but I wonder if using `Arc<Error>` vs a clone-able error might be cleaner? :thinking:

Reply:
I found this easier to adapt from the users point of view since it's basically an error pass-through.